### ISSUE TITLE
[BUG]: PyCaret does not inform shap for Shapley value calculations that independent variables had been normalized when using 'normalize=True' in data setup for regression.

### ISSUE THREAD

**Issue by DataInsightsCT** [pycaret/pycaret#3724]
### pycaret version checks

- [X] I have checked that this issue has not already been reported [here](https://github.com/pycaret/pycaret/issues).

- [X] I have confirmed this bug exists on the [latest version](https://github.com/pycaret/pycaret/releases) of pycaret.

- [ ] I have confirmed this bug exists on the master branch of pycaret (pip install -U git+https://github.com/pycaret/pycaret.git@master).

### Issue Description

When I apply PyCaret regression and use Shapley values for explainability of the model then the Shapley values show the expected relation between the dependent and independent variable(s) when 'normalize=False' in the data setup. Shapley values look different when 'normalize=True', indicating that shap was not aware of the normalization applied by PyCaret.

### Reproducible Example

```python
#!/usr/bin/env python
# coding: utf-8

# -*- coding: utf-8 -*-
"""
Created on Sun Sep  3 06:37:26 2023

Example to highlight how "normalize=True" in setup of data for regression in PyCaret affects the outcome of the Shapley values for explainability

"""

from pycaret.regression import *
from auto_shap.auto_shap import generate_shap_values


import numpy as np
import pandas as pd

import matplotlib as mpl
mpl.rcParams.update(mpl.rcParamsDefault)
import matplotlib.pyplot as plt


#Creating some starting values for dataframe
arr=np.arange(0,10.2,0.02)
noise=np.random.uniform(-10,10,size=(arr.size,))/10
arr=arr+noise
df=pd.DataFrame(arr,columns=['Column_A'])

#Creating columns of independent variables

Ba=1
Bb=1
Bc=1

df['Column_B']=Ba+Bb*df['Column_A']+Bc*df['Column_A']**2
df['Column_B']=df['Column_B']+noise*10

Ca=0
Cb=1
Cc=1

df['Column_C']=Ca+Cb*df['Column_B']+Cc*df['Column_B']**2
df['Column_C']=df['Column_C']+noise*10


#Calculating the value of the dependent variable
Da=1
Db=2
Dc=2

df['Dependent']=Da+Db*df['Column_C']+Dc*df['Column_A']**2+df['Column_B']
df['Dependent']=df['Dependent']+noise*2000

#Visualizing the synthetic data
fig, ax = plt.subplots(figsize =(10, 7))
plt.scatter(y=df['Dependent'], x=df['Column_B'])
plt.xticks(rotation=45,fontsize=8)
plt.xlabel('Column_B')
plt.ylabel('Dependent')
plt.show()


#Dropping the initial column
df=df.drop('Column_A',axis=1)
df.columns

#Setting up regression in PyCaret with 'normalize=False' in data setup

data=df.sample(frac=0.8, random_state=901).reset_index(drop=True)

exp_reg102 = setup(data = data, target = 'Dependent', session_id=651,normalize = False,remove_multicollinearity = True, multicollinearity_threshold = 0.95,imputation_type='iterative',iterative_imputation_iters=10,remove_outliers=True,n_jobs=-2) 

Random_Forest_Regressor=create_model('rf',max_depth = 10,n_estimators=50)
print(Random_Forest_Regressor)
plot_model(Random_Forest_Regressor, plot = 'feature_all')

plot_model(Random_Forest_Regressor)
plot_model(Random_Forest_Regressor, plot = 'error')


#Sampling data for calculating the Shapley values
import shap
from shap.plots import _waterfall
shap_train=get_config(variable="X_train") #Better to take it after imputing missing values?
shap_train
len(shap_train)

X_importance=shap_train.sample(50)

#Calculating the Shapley values
shap_values_df, shap_expected_value, global_shap_df = generate_shap_values(Random_Forest_Regressor, X_importance, n_jobs=20)


shap_values_df_array=shap_values_df.to_numpy()
shap.summary_plot(shap_values_df_array, features=X_importance, plot_type='bar')
shap.dependence_plot("Column_B", shap_values_df_array, X_importance, interaction_index="Column_C")
#The shape looks like how Dependent variable depends on Column_B



#Setting up regression in PyCaret with 'normalize=True' in data setup
exp_reg102 = setup(data = data, target = 'Dependent', session_id=651,normalize = True,remove_multicollinearity = True, multicollinearity_threshold = 0.95,imputation_type='iterative',iterative_imputation_iters=10,remove_outliers=True,n_jobs=-2) 

Random_Forest_Regressor=create_model('rf',max_depth = 10,n_estimators=50)
print(Random_Forest_Regressor)
plot_model(Random_Forest_Regressor, plot = 'feature_all')

plot_model(Random_Forest_Regressor)
plot_model(Random_Forest_Regressor, plot = 'error')

#Sampling data for calculating the Shapley values
import shap
from shap.plots import _waterfall
shap_train=get_config(variable="X_train") #Better to take it after imputing missing values?
shap_train
len(shap_train)

X_importance=shap_train.sample(50)

#Calculating the Shapley values
shap_values_df, shap_expected_value, global_shap_df = generate_shap_values(Random_Forest_Regressor, X_importance, n_jobs=20)

shap_values_df_array=shap_values_df.to_numpy()
shap.summary_plot(shap_values_df_array, features=X_importance, plot_type='bar')
shap.dependence_plot("Column_B", shap_values_df_array, X_importance, interaction_index="Column_C")
#The shape does not look like how Dependent variable depends on Column_B; It does not realize that the independent variables had been normalized in regression model.
```

### Expected Behavior

#Setting up regression in PyCaret with 'normalize=False' in data setup

data=df.sample(frac=0.8, random_state=901).reset_index(drop=True)

exp_reg102 = setup(data = data, target = 'Dependent', session_id=651,normalize = False,remove_multicollinearity = True, multicollinearity_threshold = 0.95,imputation_type='iterative',iterative_imputation_iters=10,remove_outliers=True,n_jobs=-2) 

Random_Forest_Regressor=create_model('rf',max_depth = 10,n_estimators=50)
print(Random_Forest_Regressor)
plot_model(Random_Forest_Regressor, plot = 'feature_all')

plot_model(Random_Forest_Regressor)
plot_model(Random_Forest_Regressor, plot = 'error')

#Sampling data for calculating the Shapley values
import shap
from shap.plots import _waterfall
shap_train=get_config(variable="X_train") #Better to take it after imputing missing values?
shap_train
len(shap_train)

X_importance=shap_train.sample(50)

#Calculating the Shapley values
shap_values_df, shap_expected_value, global_shap_df = generate_shap_values(Random_Forest_Regressor, X_importance, n_jobs=20)

shap_values_df_array=shap_values_df.to_numpy()
shap.summary_plot(shap_values_df_array, features=X_importance, plot_type='bar')
shap.dependence_plot("Column_B", shap_values_df_array, X_importance, interaction_index="Column_C")
#The shape looks like how Dependent variable depends on Column_B

### Actual Results

```python-traceback
#Setting up regression in PyCaret with 'normalize=True' in data setup
exp_reg102 = setup(data = data, target = 'Dependent', session_id=651,normalize = True,remove_multicollinearity = True, multicollinearity_threshold = 0.95,imputation_type='iterative',iterative_imputation_iters=10,remove_outliers=True,n_jobs=-2) 

Random_Forest_Regressor=create_model('rf',max_depth = 10,n_estimators=50)
print(Random_Forest_Regressor)
plot_model(Random_Forest_Regressor, plot = 'feature_all')

plot_model(Random_Forest_Regressor)
plot_model(Random_Forest_Regressor, plot = 'error')

#Sampling data for calculating the Shapley values
import shap
from shap.plots import _waterfall
shap_train=get_config(variable="X_train") #Better to take it after imputing missing values?
shap_train
len(shap_train)

X_importance=shap_train.sample(50)

#Calculating the Shapley values
shap_values_df, shap_expected_value, global_shap_df = generate_shap_values(Random_Forest_Regressor, X_importance, n_jobs=20)

shap_values_df_array=shap_values_df.to_numpy()
shap.summary_plot(shap_values_df_array, features=X_importance, plot_type='bar')
shap.dependence_plot("Column_B", shap_values_df_array, X_importance, interaction_index="Column_C")
#The shape does not look like how Dependent variable depends on Column_B; It does not realize that the independent variables had been normalized in regression model.
```

### Installed Versions

System:
    python: 3.9.16 | packaged by conda-forge | (main, Feb  1 2023, 21:28:38) [MSC v.1929 64 bit (AMD64)]

PyCaret required dependencies:
                 pip: 23.0.1
          setuptools: 67.6.0
             pycaret: 3.0.2
             IPython: 8.11.0
          ipywidgets: 7.7.0
                tqdm: 4.64.1
               numpy: 1.23.5
              pandas: 1.5.3
              jinja2: 3.1.2
               scipy: 1.10.1
              joblib: 1.2.0
             sklearn: 1.2.2
                pyod: 1.0.9
            imblearn: 0.10.1
   category_encoders: 2.6.1
            lightgbm: 3.3.5
               numba: 0.56.4
            requests: 2.28.2
          matplotlib: 3.7.1
          scikitplot: 0.3.7
         yellowbrick: 1.4
              plotly: 5.8.0
             kaleido: 0.2.1
         statsmodels: 0.13.5
              sktime: 0.17.0
               tbats: 1.1.3
            pmdarima: 2.0.3
              psutil: 5.9.4

PyCaret optional dependencies:
                shap: 0.41.0
           interpret: 0.4.2
                umap: 0.5.3
    pandas_profiling: 4.2.0
  explainerdashboard: 0.4.2.2
             autoviz: 0.1.720
           fairlearn: 0.7.0
             xgboost: 1.6.1
            catboost: 1.2
              kmodes: 0.12.2
             mlxtend: 0.22.0
       statsforecast: 1.5.0
        tune_sklearn: 0.4.3
                 ray: 1.12.0
            hyperopt: 0.2.7
              optuna: 3.2.0
               skopt: 0.9.0
              mlflow: 1.30.1
              gradio: 3.35.2
             fastapi: 0.97.0
             uvicorn: 0.22.0
              m2cgen: 0.10.0
           evidently: 0.2.8
               fugue: 0.8.5
           streamlit: Not installed
             prophet: Not installed

**Comment by moezali1** [comment]
Closing as part of the 4.0 backlog cleanup — see #4177.

PyCaret 4.0 is a ground-up rewrite (OOP-only API, sklearn-native engine, Python 3.11+, sklearn ≥ 1.7) and almost all pre-4.0 issues either reference removed APIs (`setup()`, `plot_model`, `evaluate_model`, `interpret_model`, `automl`, the legacy time-series module, the Streamlit dashboard, …) or no longer reproduce on the new code path.

**If this still affects you in 4.0, please re-file as a new issue** with:

- the experiment constructor call (`Experiment(...)`),
- the operation that fails,
- the full traceback,
- your Python / sklearn / pycaret versions.

Resources for 4.0:

- Docs: https://pycaret.org/docs
- Migrate from 3.x: https://pycaret.org/docs/guides/migrating-from-3-x
- API reference: https://pycaret.org/reference
- FAQ: https://pycaret.org/docs/resources/faqs

Thanks for filing — and apologies for the long latency. The repo had drifted; we're starting clean.
